### DISCUSSION TITLE
S3TransferManager "no_proxy" support in -PREVIEW

### DISCUSSION THREAD

**Discussion by poconnor-lab49** [aws/aws-sdk-java-v2#3632] [Q&A]
It looks like the `S3TransferManager` from software.amazon.awssdk:s3-transfer-manager automatically picks up proxy configurations.

```log
[DEBUG] [2022-12-15T19:23:05Z] [00007f3145ffb6c0] [http-connection] - HTTP_PROXY environment found
[INFO] [2022-12-15T19:23:05Z] [00007f3145ffb6c0] [http-connection] - (STATIC) Connecting to "127.0.0.1" via proxy "foo.bar.baz"
```

Is there a way to disable this, whether using environment variables or directly setting the proxy configuration like with the `S3Client`?

```java
S3Client.builder()
      .httpClientBuilder(ApacheHttpClient.builder()
          .proxyConfiguration(ProxyConfiguration.builder()
              .addNonProxyHost("localhost")
              .build())
        );
```

Hoping I've just missed some documentation, but I don't see anything like this in `2.18.38-PREVIEW`

Thanks!

**Comment by zoewangg** [comment]
Hi @poconnor-lab49 thanks for reaching out, proxy is not supported at this moment, tracking in https://github.com/aws/aws-sdk-java-v2/issues/3262

**Reply by poconnor-lab49** [reply]
Thanks for the help! I've added a feature/bug request in [aws-c-http](https://github.com/awslabs/aws-c-http) which is where I _think_ the environment variable detection is happening for proxies: https://github.com/awslabs/aws-c-http/issues/413

If that is not the root cause, please let me know where the appropriate repo is to request the change.
